pkg/client/delete.go

```diff
@@ -33,46 +35,84 @@ const (
 	e2eNamespacePrefix = "e2e-"
 )
 
+var (
+	pollFreq = 5 * time.Second
+)
+
+// Delete removes all the resources that Sonobuoy had created including
+// its own namespace, cluster roles/bindings, and optionally e2e scoped
+// namespaces.
 func (c *SonobuoyClient) Delete(cfg *DeleteConfig) error {
 	client, err := c.Client()
 	if err != nil {
 		return err
 	}
 
-	if err := cleanupNamespace(cfg.Namespace, client); err != nil {
+	conditions := []wait.ConditionFunc{}
+	nsCondition, err := cleanupNamespace(cfg.Namespace, client)
```

Comment:
I wrote the logic here so that the method signatures of each of these cleanup steps is responsible for creating (and returning) their own "wait" function. This is so that the logic for the action is right next to the logic regarding what needs to be waited on.

Maybe it would be better to define these as unexported functions though (outside of the functions) so that its a bit easier to test? I do wish these had more test coverage; I think that is the main todo before merge. Good to test manually a.t.m. but needs some automated testing.

Reply:
Were you going to add tests? I don't see any added.

Reply:
So when we did a brief code review in person I pointed out how, once you extract the boilerplate of getting clients and making API calls, the actual logic of these wait methods is akin to `if empty` or something similarly trivial. At the time I thought you were saying you were OK without adding tests; you want me to do that though?

I could refactor it to try and include tests for each of the wait conditions and then for the "all conditions" condition.

Reply:
Ok cool, was just going off of your comments inline. No problem.